### ISSUE TITLE
Python Windows error "hi = thdata[96][128][0]"

### ISSUE THREAD

**Issue by Riska2390** [leswright1977/PyThermalCamera#15]
hi = thdata[96][128][0]
IndexError: index 96 is out of bounds for axis 0 with size 0

**Comment by m-riley04** [comment]
Also getting this error

**Comment by stsypanov** [comment]
Is is an issue with OTG cable?

I get this with DIY USB-Type-C-female-to-micro-USB-male cable. Nothing special, just +5V, GND, D+, D-

**Comment by m-riley04** [comment]
I actually was able to fix this issue through a large fork I made. It essentially refactors the entire program to be more modular as well.

IIRC, the issue was during the RGB conversion check with Raspberry Pi's. After removing that entire block, it works without a hitch every time.

It even directly addresses this issue as well, since it's rebuilt around being Windows-based.

Fork repo: https://github.com/m-riley04/PyThermalCamera-Windows

**Comment by stsypanov** [comment]
@m-riley04 Many thanks! From your code I see that the fix was introduced in the very first [commit](https://github.com/leswright1977/PyThermalCamera/commit/eb02b326655531c15f12502e6da481987e335e77), is my guess correct?

**Comment by m-riley04** [comment]
Yup! I commented out the lines setting the `cv2.CAP_PROP_CONVERT_RGB` property to 0.0 (or `False`). After doing so, it seemed to work perfectly on Windows

**Comment by stsypanov** [comment]
@m-riley04 thanks, using your code I've finally elaborated the solution for Raspberry Pi Zero 2W. I think I owe you a beer :)
@Riska2390 FYI

![Image](https://github.com/user-attachments/assets/d5a4e509-bedd-4281-b0ff-e3fc8d7be487)
